test/Test/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Test.Database.LSMTree.Internal.RawPage (
+    -- * Main test tree
+    tests,
+) where
+
+import qualified Data.ByteString.Short as SBS
+import           Data.Primitive.ByteArray (ByteArray (..), byteArrayFromList)
+import qualified Data.Vector.Primitive as V
+import           Data.Word (Word16)
+import           GHC.Word (byteSwap16)
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.HUnit (testCase, (@=?))
+
+import           Database.LSMTree.Internal.RawPage
+import           FormatPage (Key (..), Operation (..), PageLogical (..),
+                     Value (..), encodePage, serialisePage)
+
+tests :: TestTree
+tests = testGroup "Database.LSMTree.Internal.RawPage"
+    [ testCase "empty" $ do
+        -- directory:
+        -- * 0 keys,
+        -- * 0 blobs
+        -- * key's offset offset at 8 bytes (as bitmaps are empty)
+        -- * nil spare
+        --
+        -- and
+        -- offset past last value.
+        let bytes :: [Word16]
+            bytes = [0, 0, 8, 0, 10]
+
+        let page = RawPage (byteArrayFromList bytes)
+
+        page @=? toRawPage (PageLogical [])
+        rawPageNumKeys page @=? 0
+        rawPageNumBlobs page @=? 0
+        rawPageKeyOffsets page @=? V.fromList []
+
+    , testCase "single-insert" $ do
+        let bytes :: [Word16]
+            bytes =
+                [ 1, 0, 24, 0             -- directory
+                , 0, 0, 0, 0, 0, 0, 0, 0  -- ...
```

Comment:
These should be 16 bytes, 8 for the blobref indicators and 8 for the operation types

Reply:
Never mind, I missed that these were `Word16`s